### PR TITLE
VET-71 Modify Client - frontend

### PR DESCRIPTION
**Pull request by krawczykaghust**
Main goal of this task was to be able to update client data from frontend view.

This was done by using few things:

- use updateClientfunction from the backend clientController.js file - for this purpose use a body-parser library
- create several functionalities to be able to use modify on each client
- download and use react-boostrap library to be able to work with Modal component to display the form when clicking the button (in the future will be made as pop-up, now it shows and brings the user to the end of the page)

### CODE REVIEW

File: frontend/src/components/ClientDetails.js [hunk 1]
```diff
@@ -42,19 +87,59 @@ const ClientDetails = ({ client }) => {
...
+                        <div className="modal-form-group">
+                            <label htmlFor="imie">Imię</label>
+                            <input type="text" className="modal-form-control" name="imie" value={formData.imie} onChange={handleChange} />
+                        </div>
+                        <div className="modal-form-group">
+                            <label htmlFor="nazwisko">Nazwisko</label>
+                            <input type="text" className="modal-form-control" name="nazwisko" value={formData.nazwisko} onChange={handleChange} />
+                        </div>
+                        <div className="modal-form-group">
+                            <label htmlFor="numer_konta">Numer Konta</label>
```
Comment (MichSawi) [inline frontend/src/components/ClientDetails.js]: No capital letter needed in Konta

File: frontend/src/context/ClientContext.js [hunk 2]
```diff
@@ -7,6 +7,14 @@ export const clientsReducer = (state, action) => {
...
         case 'CREATE_CLIENT':return {clients: [action.payload, ...state.clients]}
         case 'DELETE_CLIENT': return {clients: state.clients.filter((c) => c._id !== action.payload._id)}
+        case 'UPDATE_CLIENTS': 
+        const updatedClients = state.clients.map(client => {
+            if (client._id === action.payload._id) {
+            return action.payload;
+            }
+            return client;
+        });
+        return { clients: updatedClients };
```
Comment (MichSawi) [inline frontend/src/context/ClientContext.js]: Space after bracket and before bracket not needed

File: frontend/src/components/ClientDetails.js [hunk 3]
```diff
@@ -1,13 +1,58 @@
...
+    const handleChange = (event) => {
+        setFormData({
+            ...formData,
+            [event.target.name]: event.target.value
+        })
+    }
+
+    const handleSubmitModal = async () => {
+        if (!user) {
+            return
```
Comment (MichSawi) [inline frontend/src/components/ClientDetails.js:32]: Semicolon after return

**Review by MichSawi — CHANGES_REQUESTED** [review]
Few minor style changes

File: frontend/src/components/ClientDetails.js (hunk 3 above)
Comment (krawczykaghust) [reply frontend/src/components/ClientDetails.js:32]: Refactored

File: frontend/src/context/ClientContext.js (hunk 2 above)
Comment (krawczykaghust) [reply frontend/src/context/ClientContext.js]: Refactored

File: frontend/src/components/ClientDetails.js (hunk 1 above)
Comment (krawczykaghust) [reply frontend/src/components/ClientDetails.js]: Refactored

**Review by MichSawi — APPROVED** [review]
Changes approved

**Pull request merged**